### ISSUE TITLE
The ScriptsMiddleware(s) should be replaced by endpoints

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/OrchardCore#15629]
No need to create a middleware that is impacting the stack size to serve script files. We can register the endpoint routes directly and invoke a delegate.

These methods should also not do string concatenation to render the files. They could precompute the utf8 string (`private static ReadOnlySpan<byte> => "value"u8;`) and call `WriteAsync()` for each segment instead.

**Comment by sebastienros** [comment]
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Liquid/ScriptsMiddleware.cs

https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Facebook/ScriptsMiddleware.cs

**Comment by Piedone** [comment]
The [endpoints in the Content module](https://github.com/OrchardCMS/OrchardCore/tree/c75ed73353d0744b1855e2007862bf89ad42d493/src/OrchardCore.Modules/OrchardCore.Contents/Endpoints/Api) can serve as examples.

**Comment by DonaldDWebster** [comment]
Is this issue still open? I was thinking about picking it up as a first issue if so, I wanted to check first since I do not see any references to OrchardCore.Liquid.ScriptsMiddleWare

Forgive my ignorance and thank you for the help.

**Comment by hishamco** [comment]
You can go for it

**Comment by Piedone** [comment]
Yep, thank you! Be sure to check out https://docs.orchardcore.net/en/latest/guides/contributing/contributing-code/ first.

**Comment by DonaldDWebster** [comment]
My progress has been slow, I don't think I can complete it in a reasonable amount of time 😅. My sincere apologies.

**Comment by Piedone** [comment]
Perhaps try to follow the pattern of https://github.com/OrchardCMS/OrchardCore/pull/15294. This should be more or less copy-pasting to endpoints from the middlewares.

**Comment by jbytes1027** [comment]
@DonaldDWebster are you still planning on working on this? I'm happy to finish this for you.

**Comment by DonaldDWebster** [comment]
> @DonaldDWebster are you still planning on working on this? I'm happy to finish this for you.

Please feel free to pick it up
